ale/base/type_sensor.py

```diff
@@ -362,3 +368,125 @@ def line_times(self):
         : array
         """
         raise NotImplementedError
+
+
+class Cahvor():
+    """
+    Mixin for largely ground based sensors to add an
+    extra step in the frame chain to go from ground camera to
+    the Camera
+    """
+
+    @property
+    def cahvor_camera_dict(self):
+        """
+        This function extracts and returns the elements for the
+        CAHVOR camera model from a concrete driver as a dictionary.
+        See the MSL MASTCAM Cahvor, Framer, Pds3Label, NaifSpice, Driver
+        """
+        raise NotImplementedError
+
+    @property
+    def cahvor_rotation_matrix(self):
+        """
+        Computes the cahvor rotation matrix for the instrument to Rover frame
+
+        Returns
+        -------
+        : array
+          Rotation Matrix as a 2D numpy array
+        """
+        if not hasattr(self, "_cahvor_rotation_matrix"):
+            h_c = np.dot(self.cahvor_camera_dict['A'], self.cahvor_camera_dict['H'])
+            h_s = np.linalg.norm(np.cross(self.cahvor_camera_dict['A'], self.cahvor_camera_dict['H']))
```

Comment:
We compute this a few times in different places. It may be helpful to pull it into a separate function for readability.